### PR TITLE
rationalise UnsafeMemory.copyMemory functions, optimise performance, improve test coverage

### PR DESCRIPTION
**Pull request by JerryShea**
Confirmed now that this works with build-all. I had to restore the loop for some on heap Object copies

### CODE REVIEW

**Review by RobAustin — APPROVED** [review]

**Review by rogersimmons — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/core/UnsafeMemory.java [hunk 1]
```diff
@@ -570,61 +547,63 @@ public void copyMemory(long fromAddress, long address, long length) {
...
-
+            UNSAFE.copyMemory(fromAddress, address, length);
         } else {
             copyMemory0(null, fromAddress, null, address, length);
         }
     }
 
     @Override
-    public void copyMemory(byte[] bytes, int offset, Object obj2, long offset2, int length) {
+    public void copyMemory(byte[] bytes, int offset, @Nullable Object obj2, long offset2, int length) {
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/core/UnsafeMemory.java]: I feel like the names `source` and `destination` and `sourceOffset` and `destinationOffset` might be clearer for all these methods.

Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/core/UnsafeMemory.java]: (or similar)

**Review by nicktindall — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/core/UnsafeMemory.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/core/UnsafeMemory.java]: yep - agree - I will change. I know you don't like short names, but how about src and dest?

Comment (minborg) [reply src/main/java/net/openhft/chronicle/core/UnsafeMemory.java]: I prefer `source` over `src` etc.

**Pull request merged**